core/src/HIP/Kokkos_HIP.cpp

```diff
@@ -75,6 +75,14 @@ void HIP::impl_initialize(InitializationSettings const& settings) {
   }
 #endif
 #ifdef KOKKOS_ARCH_AMD_GFX942_APU
+  if (!Impl::HIPInternal::singleton().is_xnack_enabled()) {
+    Kokkos::abort(
```

Comment:
Why do we want this to be an abort rather than just a warning?

Reply:
> In the Kokkos developer meeting 2024-12-11, it was decided that it was reasonable to require xnack for the MI300A, so error out of HIP initialize if its not enabled.

Reply:
@dalg24, @JBludau, and @Rombur discussed this and we feel that since standard uses of the HIP backend will work either way on MI300A, it's better just to warn. The only problematic scenario we're aware of is if you wrap `new()` inside a `SharedSpace` - if you just ask for a `View<SharedSpace>` it's the managed memory allocator and everything will work even if `HSA_XNACK=0` or unset (see tests in https://github.com/kokkos/kokkos/pull/7567#issuecomment-2537052491)